packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+				},
+			});
+			expect(eth2.transactionBlockTimeout).toBe(120);
+		});
+		it('transactionConfirmationBlocks', () => {
+			// default
+			expect(web3Eth.transactionConfirmationBlocks).toBe(24);
+
+			// after set
+			web3Eth.setConfig({
+				transactionConfirmationBlocks: 3,
+			});
+			expect(web3Eth.transactionConfirmationBlocks).toBe(3);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionConfirmationBlocks: 4,
+				},
+			});
+			expect(eth2.transactionConfirmationBlocks).toBe(4);
+			// implementation tested here ./watch_transaction.test.ts
+		});
+		it('transactionPollingInterval and transactionPollingTimeout', () => {
+			// default
+			expect(web3Eth.transactionPollingInterval).toBe(1000);
+			expect(web3Eth.transactionPollingTimeout).toBe(750);
+
+			// after set
+			web3Eth.setConfig({
+				transactionPollingInterval: 3,
+				transactionPollingTimeout: 10,
+			});
+			expect(web3Eth.transactionPollingInterval).toBe(3);
+			expect(web3Eth.transactionPollingTimeout).toBe(10);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionPollingInterval: 400,
+					transactionPollingTimeout: 10,
+				},
+			});
+			expect(eth2.transactionPollingInterval).toBe(400);
+			expect(eth2.transactionPollingTimeout).toBe(10);
+		});
+		it('transactionReceiptPollingInterval and transactionConfirmationPollingInterval', () => {
+			// default
+			expect(web3Eth.transactionReceiptPollingInterval).toBeNull();
+			expect(web3Eth.transactionConfirmationPollingInterval).toBeNull();
+
+			// after set
+			web3Eth.setConfig({
+				transactionReceiptPollingInterval: 3,
+				transactionConfirmationPollingInterval: 10,
+			});
+			expect(web3Eth.transactionReceiptPollingInterval).toBe(3);
+			expect(web3Eth.transactionConfirmationPollingInterval).toBe(10);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionReceiptPollingInterval: 400,
+					transactionConfirmationPollingInterval: 10,
+				},
+			});
+			expect(eth2.transactionReceiptPollingInterval).toBe(400);
+			expect(eth2.transactionConfirmationPollingInterval).toBe(10);
+		});
+		it('blockHeaderTimeout', () => {
+			// default
+			expect(web3Eth.blockHeaderTimeout).toBe(10);
+
+			// after set
+			web3Eth.setConfig({
+				blockHeaderTimeout: 3,
+			});
+			expect(web3Eth.blockHeaderTimeout).toBe(3);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					blockHeaderTimeout: 4,
+				},
+			});
+			expect(eth2.blockHeaderTimeout).toBe(4);
+		});
+		it('maxListenersWarningThreshold', () => {
+			// default
+			expect(web3Eth.maxListenersWarningThreshold).toBe(100);
+
+			// after set
+			web3Eth.setConfig({
+				maxListenersWarningThreshold: 3,
+			});
+			expect(web3Eth.maxListenersWarningThreshold).toBe(3);
```

Comment:
could you attach 4 listeners and capture warning

Reply:
maxListenersWarningThreshold is not implemented yet https://github.com/ChainSafe/web3.js/issues/5087

Reply:
@jdevcs I've resolved this conversation.